software/controller/lib/hal/spi_chain.tpp

```diff
@@ -192,7 +192,21 @@ bool DaisyChain<MaxSlaves, QueueLength>::SendRequest(const Request &request, siz
   }
   // Ensure thread safety as this function might be called from a timer interrupt
   // as well as the main loop.
-  BlockInterrupts block;
+  if(!Interrupts::singleton().InInterruptHandler()) {
+    // We don't block interrupts if we are already in an interrupt handler. This means that we are
+    // only allowing a timer interrupt to call this function: our high priority task. No other
+    // interrupt handler should send SPi requests.
+    // It also means that high priority task needs to be finished before the requests it sends are
+    // fully transmitted (should be OK as the DMA interrupt management takes longer than the high
+    // priority task: by the time we get the first DMA interrupt, the high priority task is already
+    // finished - as of this writing, but we don't expect high priority task to get much
+    // bigger that it already is).
+    // This is a (hopefully temporary) solution to issue 1298, and should be dealt with more nicely
+    // once we have implemented proper framing for the debug interface, which we will then be able
+    // to run with DMA.
```

Comment:
Maybe call out the ticket for DMA implemetation? And leave a `todo` for changing this behavior once that is done.